src/RESTDataSource.ts

```diff
@@ -171,13 +172,20 @@ export type RequestDeduplicationPolicy =
   // the deduplication store.)
   | { policy: 'do-not-deduplicate'; invalidateDeduplicationKeys?: string[] };
 
+export interface Logger {
+  info(msg: string): void;
+  error(error: any, ...data: any[]): void;
+}
+
 export abstract class RESTDataSource {
   protected httpCache: HTTPCache;
   protected deduplicationPromises = new Map<string, Promise<any>>();
   baseURL?: string;
+  logger: Logger;
 
   constructor(config?: DataSourceConfig) {
     this.httpCache = new HTTPCache(config?.cache, config?.fetch);
+    this.logger = config?.logger ? config.logger : console;
```

Comment:
Nit: just a preference here for nullish coalesce
```suggestion
    this.logger = config?.logger ?? console;
```